validphys2/src/validphys/theorycovariance/construction.py

```diff
@@ -199,6 +225,234 @@ def covmat_9pt(name1, name2, deltas1, deltas2):
+        start_proc[name] = running_index
+        running_index += size
+
+        # Compute shifts only for a subset of processes
+        if name in filtered_name:
+            central = process_info.preds[name][1]
+            kin1, kin2 = process_info.data[name].T[:2]
+            if ht_pt_prescription == 5 or ht_pt_prescription == 9:
+                deltas["(+,0)"] += [central / kin2 * ht_coeff_1] # (1,0)
+                deltas["(0,+)"] += [central / kin2 * ht_coeff_2 * kin1 / (1 - kin1)] # (0,1)
+                if ht_pt_prescription == 9:
+                        deltas["(+,+)"] += [central / kin2 * (ht_coeff_1 + ht_coeff_2 * kin1 / (1 - kin1))] # (1,1)
+                        deltas["(+,-)"] += [central / kin2 * (ht_coeff_1 - ht_coeff_2 * kin1 / (1 - kin1))] # (1,-1)
+            else:
+                raise ValueError(
+                    f"The pt prescription for the HT theory covmat is not supported. "\
+                    f"Please, choose either 5 or 9."
+                )
+
+    # Construct theory covmat
+    covmats = defaultdict(list)
+    for i,proc1 in enumerate(filtered_name):
+        for j,proc2 in enumerate(filtered_name):
+            if ht_pt_prescription == 5:
+                s = np.outer(deltas["(+,0)"][i], deltas["(+,0)"][j]) + \
+                    np.outer(deltas["(0,+)"][i], deltas["(0,+)"][j])
+            elif ht_pt_prescription == 9:
+                s = 0.5 * ( np.outer(deltas["(+,0)"][i], deltas["(+,0)"][j]) + \
+                            np.outer(deltas["(0,+)"][i], deltas["(0,+)"][j]) + \
+                            np.outer(deltas["(+,+)"][i], deltas["(+,+)"][j]) + \
+                            np.outer(deltas["(+,-)"][i], deltas["(+,-)"][j]) )
+            start_locs = (start_proc[proc1], start_proc[proc2])
+            covmats[start_locs] = s
+    return covmats
+
+
+def thcov_HT_4(combine_by_type_ht, ht_coeff_1, ht_coeff_2, ht_pt_prescription = 5):
+    "Same as `thcov_HT` but implementing 5pt and 9pt prescriptions."
+
+    process_info = combine_by_type_ht
+    running_index_tot = 0
+    start_proc_by_exp = defaultdict(list)
+    deltas = defaultdict(list)
+    included_proc = ["DIS NC", "DIS CC"]
+    excluded_exp = ["NMC_NC_NOTFIXED_DW_EM-F2",
+                    "CHORUS_CC_NOTFIXED_PB_DW_NU-SIGMARED",
+                    "CHORUS_CC_NOTFIXED_PB_DW_NB-SIGMARED",
+                    "NUTEV_CC_NOTFIXED_FE_DW_NU-SIGMARED",
+                    "NUTEV_CC_NOTFIXED_FE_DW_NB-SIGMARED"]
+    filtered_proc = {k : v for k, v in filter(lambda t: t[0] in included_proc , process_info.preds.items())}
+
+    # ABMP parametrisation
+    x = [0.0, 0.1, 0.3, 0.5, 0.7, 0.9, 1]
+    y_2 = [0.023, -0.032, -0.005, 0.025, 0.051, 0.003, 0.0]
+    y_T = [-0.319, -0.134, -0.052, 0.071, 0.030, 0.003, 0.0]
+    H_2 = scint.CubicSpline(x, y_2)
+    H_2 = np.vectorize(H_2)
+    H_T = scint.CubicSpline(x, y_T)
+    def H_L(x):
+        return np.power(x, 0.05)*(H_2(x) - H_T(x))
```

Comment:
```suggestion
        return (H_2(x) - np.power(x, 1.05)*H_T(x))
```
Looking at the paper this prefactor is there only for F_T